spring-cloud-circuitbreaker-resilience4j/src/test/java/org/springframework/cloud/circuitbreaker/resilience4j/BulkheadConfigurationTests.java

```diff
@@ -0,0 +1,181 @@
+/*
+ * Copyright 2013-2024 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.cloud.circuitbreaker.resilience4j;
+
+import io.github.resilience4j.bulkhead.BulkheadRegistry;
+import io.github.resilience4j.bulkhead.ThreadPoolBulkhead;
+import io.github.resilience4j.bulkhead.ThreadPoolBulkheadConfig;
+import io.github.resilience4j.bulkhead.ThreadPoolBulkheadRegistry;
+import org.junit.jupiter.api.Test;
+
+import org.springframework.boot.autoconfigure.EnableAutoConfiguration;
+import org.springframework.boot.test.context.runner.WebApplicationContextRunner;
+import org.springframework.context.annotation.Configuration;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+/**
+ * @author Ryan Baxter
+ */
+public class BulkheadConfigurationTests {
+
+	@Test
```

Comment:
Do we want to test these?
What is expected?

`Resilience4jBulkheadProvider::configureDefault`
`Resilience4jBulkheadProvider::configure`
`Resilience4jBulkheadProvider::addBulkheadCustomizer`
`Resilience4jBulkheadProvider::addThreadPoolBulkheadCustomizer`

Reply:
Example:

`Resilience4jBulkheadProvider::configureDefault` vs `resilience4j.threadPoolBulkHead.configs.default.queueCapacity=30`

In my opinion, when `Resilience4jBulkheadProvider::configureDefault` is used, then it take precedence over property. Perhaps same with `Resilience4jBulkheadProvider::configure` ?

Reply:
Resilience4jBulkheadProvider::configureDefault, Resilience4jBulkheadProvider::addBulkheadCustomizer, Resilience4jBulkheadProvider::addThreadPoolBulkheadCustomizer are used in Resilience4JBulkheadIntegrationTest and  and Resilience4JBulkheadAndTimeLimiterIntegrationTest.

I agree on the configureDefault and I added some tests for that.